components/metrics/metrics_service.cc

```diff
@@ -605,6 +610,30 @@ void MetricsService::OpenNewLog() {
     // We only need to schedule that run once.
     state_ = INIT_TASK_SCHEDULED;
 
+#if defined(STARBOARD)
```

Comment:
Could this be refactored to:

```
    int initialization_delay_secs = kInitializationDelaySeconds;
    
    #if defined(STARBOARD)
    const base::CommandLine* command_line =
        base::CommandLine::ForCurrentProcess();
    if (command_line->HasSwitch(switches::kInitialMetricsUploadIntervalSec)) {
      if (base::StringToInt(command_line->GetSwitchValueASCII(
                                switches::kInitialMetricsUploadIntervalSec),
                            &initialization_delay_secs)) {
        LOG(INFO) << "Metrics initial delay overriden to: "
                  << initialization_delay_secs;
      }
    }
    #endif
    base::SequencedTaskRunnerHandle::Get()->PostDelayedTask(
        FROM_HERE,
        base::BindOnce(&MetricsService::StartInitTask,
                       self_ptr_factory_.GetWeakPtr()),
        base::TimeDelta::FromSeconds(initialization_delay_secs));

    base::SequencedTaskRunnerHandle::Get()->PostDelayedTask(
        FROM_HERE,
        base::BindOnce(&MetricsService::PrepareProviderMetricsTask,
                       self_ptr_factory_.GetWeakPtr()),
        base::TimeDelta::FromSeconds(2 * initialization_delay_secs));
```

to avoid repeated code